### PR TITLE
Sscs 3050 postcode venue robotics

### PR DESCRIPTION
**Pull request by chrisdavecm**

### CODE REVIEW

File: src/main/java/uk/gov/hmcts/sscs/controller/PostCodeController.java [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+public class PostCodeController {
+    private static final org.slf4j.Logger LOG = getLogger(PostCodeController.class);
+
+    @Autowired
+    private AirLookupService airLookupService;
+
+    @ApiOperation(value = "getRegionalCentre",
+            notes = "Returns the regional centre given the appellants full post code or out code",
+            response = String.class)
+    @ApiResponses(value = {@ApiResponse(code = 200, message = "Regional Centre", response = String.class)})
```
Comment (chrisgrimble) [inline src/main/java/uk/gov/hmcts/sscs/controller/PostCodeController.java]: 404 if we can't find the postcode

File: src/main/java/uk/gov/hmcts/sscs/json/RoboticsJsonMapper.java [hunk 2]
```diff
@@ -35,11 +35,11 @@ public JSONObject map(RoboticsWrapper wrapper) {
         return obj;
     }
 
-    private static JSONObject buildAppealDetails(JSONObject obj, SyaCaseWrapper appeal) {
+    private static JSONObject buildAppealDetails(JSONObject obj, SyaCaseWrapper appeal, String venueName) {
         obj.put("caseCode", "002DD");
         obj.put("appellantNino", appeal.getAppellant().getNino());
         //FIXME: To be implemented at a future date as part of another ticket
-        obj.put("appellantPostCode", "Bedford");
+        obj.put("appellantPostCode", venueName);
```
Comment (chrisgrimble) [inline src/main/java/uk/gov/hmcts/sscs/json/RoboticsJsonMapper.java]: The fixme comment can now go can't it?

File: src/main/java/uk/gov/hmcts/sscs/service/SubmitAppealService.java [hunk 3]
```diff
@@ -67,22 +74,45 @@
...
             sendJsonByEmail(appeal.getAppellant(), roboticsJson);
         }
     }
 
+    protected String getFirstHalfOfPostcode(String postcode) {
+        if (postcode != null && postcode.length() > 3) {
+            return postcode.substring(0, postcode.length() - 3).trim();
+        }
+        return "";
+    }
```
Comment (chrisgrimble) [inline src/main/java/uk/gov/hmcts/sscs/service/SubmitAppealService.java]: This is also in AirLookupService maybe we can pull it out to a util

**Comment by chrisdavecm** [comment]
This was superceded by https://github.com/hmcts/tribunals-case-api/pull/181

**Pull request closed without merging**